src/Sylius/Bundle/OrderBundle/NumberGenerator/SequentialOrderNumberGenerator.php

```diff
@@ -92,11 +123,23 @@ private function getSequence()
         /** @var OrderSequenceInterface $sequence */
         $sequence = $this->sequenceRepository->findOneBy([]);
 
-        if (null === $sequence) {
-            $sequence = $this->sequenceFactory->createNew();
-            $this->sequenceRepository->add($sequence);
+        if (null !== $sequence) {
+            return $sequence;
         }
 
+        $sequence = $this->sequenceFactory->createNew();
+        $this->sequenceManager->persist($sequence);
+
         return $sequence;
     }
+
+    private function openSequenceManagerIfClosed()
+    {
+        if (!$this->sequenceManager->isOpen()) {
+            $this->sequenceManager = EntityManager::create(
```

Comment:
This smells a bit... we are creating new entity manager here, not really opening the existing one.